src/Microsoft.Health.Fhir.Api/Controllers/ExportController.cs

```diff
@@ -0,0 +1,134 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+using System;
+using System.Collections.Generic;
+using System.Net;
+using EnsureThat;
+using Hl7.Fhir.Model;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Api.Features.ActionResults;
+using Microsoft.Health.Fhir.Api.Features.Filters;
+using Microsoft.Health.Fhir.Api.Features.Routing;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Features.Context;
+
+namespace Microsoft.Health.Fhir.Api.Controllers
+{
+    [ServiceFilter(typeof(OperationOutcomeExceptionFilterAttribute))]
+    public class ExportController : Controller
+    {
+        private readonly ILogger<ExportController> _logger;
+        private readonly IFhirRequestContextAccessor _fhirRequestContextAccessor;
+        private readonly ExportConfiguration _exportConfig;
+
+        public ExportController(
+            ILogger<ExportController> logger,
+            IFhirRequestContextAccessor fhirRequestContextAccessor,
+            IOptions<ExportConfiguration> exportConfig)
+        {
+            EnsureArg.IsNotNull(logger, nameof(logger));
+            EnsureArg.IsNotNull(fhirRequestContextAccessor, nameof(fhirRequestContextAccessor));
+            EnsureArg.IsNotNull(exportConfig, nameof(exportConfig));
+
+            _logger = logger;
+            _fhirRequestContextAccessor = fhirRequestContextAccessor;
+            _exportConfig = exportConfig.Value;
+        }
+
+        [HttpGet]
+        [Route(KnownRoutes.Export)]
+        [ValidateExportHeadersFilter]
+        public IActionResult Export()
+        {
+            HttpStatusCode returnCode;
+            OperationOutcome result;
+
+            if (_exportConfig.SupportsExport)
+            {
+                result = GenerateOperationOutcome(
```

Comment:
We should just be able to throw the appropriate exceptions here instead of newing up fhirresults and outcomes. 

Doing this should take care of setting the appropriate status code and operation outcomes.